Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -295,20 +295,20 @@ public final class KafkaConsumer: Sendable, Service {
     private func assign(
```

Comment:
I believe that `assign` and especially `subscribe` should be public.
There could be a valid scenario when topics are not known (or not all topics/partitions known) at the point of consumer creation. If method is private, I see two possible solutions:
1. Re-create whole `KafkaConsumer` with newly known topics -> slow
2. Create additional `KafkaConsumer` for new topics -> thread explosion by librdkafka 

If allow to use `subscribe/unsubscribe`, it would be possible to share one consumer across multiple topics/partitions. Otherwise, if topics are unknown or partially unknown at startup, it would require to always create consumer per new topic or re-create it.

Reply:
I understand the need for this but I am not sure if we should provide those and if we can do so safely. The problem is that if we allow manual change of topics/partitions then we might still have messages buffered from the previous and it might be confusing for the user.

At least we should track this in a separate issue and not make it part of this PR.

Reply:
There are some thought on that.
I believe that your comment is more related to unassign and unsubscribe. However, `assign/incremental_assign/incremental_unassign` are used for rebalance. Which in its turn used pretty widely. Therefore, if it is planned, it might be nice to keep it in mind (probably it be relates to other comment for two streams for events/messages if sync is important to do from the library perspective). 

Anyway I was more concerned regarding `subscribe` method that might be helpful and should not contradict with safety.

Reply:
Yes definitely! Feel free to create an issue for that so we can kick off a more detailed discussion there!

Reply:
After giving this more thought we 100% cannot expose a subscribe and assign with the current APIs. This would mean users would expect to iterate the `messages` async sequence more than once (a new iteration after each subscribe/assign) call which is not allowed right now.

Reply:
What I would like to see is a concrete use-case where this is required which is not motivated by the performance of creating new consumers but rather from business logic needs.

Reply:
I am pretty sure that all business logic needs can be covered by creating/removing consumers.